internal/query/relations.go

```diff
@@ -631,13 +820,50 @@ func (e *engine) ListRoles(ctx context.Context, resource types.Resource) ([]type
 		},
 	}
 
+	dbDone := make(chan struct{}, 1)
+
+	var (
+		dbRoles []*database.Role
+		dbErr   error
+	)
+
+	go func() {
+		defer close(dbDone)
+
+		dbRoles, dbErr = e.db.ListResourceRoles(ctx, resource.ID)
+	}()
+
 	relationships, err := e.readRelationships(ctx, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	out := relationshipsToRoles(relationships)
 
+	<-dbDone
+
+	if dbErr != nil {
+		if !errors.Is(dbErr, database.ErrNoRoleFound) {
+			return nil, dbErr
+		}
+	} else {
+		rolesByID := make(map[string]*database.Role, len(dbRoles))
+		for _, role := range dbRoles {
+			rolesByID[role.ID.String()] = role
+		}
+
+		for i, role := range out {
+			if dbRole, ok := rolesByID[role.ID.String()]; ok {
+				role.Name = dbRole.Name
+				role.Creator = dbRole.CreatorID
+				role.CreatedAt = dbRole.CreatedAt
+				role.UpdatedAt = dbRole.UpdatedAt
+
+				out[i] = role
+			}
+		}
+	}
```

Comment:
Have we determined that the time cost of getting roles from both CRDB and SpiceDB is worth the complexity of adding concurrency here?